### ISSUE TITLE
created ARRAYS are not ordered

### ISSUE THREAD

**Issue by manticore-projects** [duckdb/duckdb#11484]
### What happens?

Creating an Array will shuffle the elements randomly, which:
a) does not match the documentation, stating: 

[Ordering](https://duckdb.org/docs/sql/data_types/array#ordering)

The ordering of ARRAY instances is defined using a lexicographical order. NULL values compare greater than all other values and are considered equal to each other.

b) becomes a problem when accessing the elements

### To Reproduce

```sql

-- repeat a few times
SELECT ARRAY
  (SELECT 1 UNION ALL
   SELECT 2 UNION ALL
   SELECT 3
   ORDER by 1) AS new_array;

/* outputs are:
┌───────────┐
│ new_array │
├───────────┤
│ [2, 3, 1] │
└───────────┘

┌───────────┐
│ new_array │
├───────────┤
│ [2, 1, 3] │
└───────────┘

┌───────────┐
│ new_array │
├───────────┤
│ [3, 1, 2] │
└───────────┘

*/

```

### OS:

Linux

### DuckDB Version:

0.10.0 .. 0.11-Snapshot

### DuckDB Client:

Java

### Full Name:

Andreas Reichel

### Affiliation:

Reichel

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by manticore-projects** [comment]
Even an `ORDER BY` clause does not help:

```sql

-- still random order
SELECT ARRAY
  (SELECT 1 UNION ALL
   SELECT 2 UNION ALL
   SELECT 3 order by 1) AS new_array;

-- although the `SetOperation` is always sorted
SELECT 1 UNION ALL
   SELECT 2 UNION ALL
   SELECT 3 order by 1;

--> strictly returns rows 1, 2, 3
```

**Comment by soerenwolfers** [comment]
Why `ORDER BY 1`? Did you try ordering by the actual column?

Also, the bit of the documentation you cite is about comparisons of different arrays, not about ordering within the array (although I agree that sorted inputs should be preserved)

**Comment by soerenwolfers** [comment]
(1) Is `ARRAY` documented anywhere? 

(2) The bit of the documentation that you quote talks about the ordering of different arrays, not the ordering of elements within an array

(3) For now, you can use `array_agg(..., ORDER BY COLUMNNAME)` (you need to specify the column name here, integer constants are treated as literals, not as column specifiers. Not sure if that should be made it's own ticket)

**Comment by manticore-projects** [comment]
> (2) The bit of the documentation that you quote talks about the ordering of different arrays, not the ordering of elements within an array

Ok :-D

```sql
SELECT ARRAY
  (SELECT ARRAY[1] UNION ALL
   SELECT ARRAY[2] UNION ALL
   SELECT ARRAY[3] order by 1) AS new_array;

/*
"new_array"
"[[1], [3], [2]]"
*/
```

**Comment by manticore-projects** [comment]
We can also close this, when it is rather a feature than a bug. I only recorded what I noticed as unexpected behavior from my own point of view.

**Comment by soerenwolfers** [comment]
I don't know if it's expected because I can't find documentation for it. 

In any case, your example still doesn't contradict the documentation, not just because the documentation doesn't talk about the ARRAY construct you're using but also because the documentation talks about how you can compare an array `a1` to an array `a2`, not about how array entries are sorted or how sorting is preserved from subqueries.

**Comment by soerenwolfers** [comment]
Practically speaking, I don't think you should worry about this and just use the regular and documented aggregate function `array_agg` or the regular constructor function `array_value` for now.

**Comment by Mytherin** [comment]
Thanks for the report! This is specifically related to using constant indexes (e.g. `ORDER BY 1`). I've pushed a fix in #11674.
